asyncPrimeSearching.cpp

```diff
@@ -26,22 +32,45 @@ struct primesInRange {
     }
 };
 
-std::vector<mpz_class> findPrimes(mpz_class start, mpz_class finish) {
-  
-    int threadTotal = std::thread::hardware_concurrency();
-    if (threadTotal == 0) return std::vector<mpz_class>{};
+
+std::vector<mpz_class> gatherResultsFromThreads(
+    std::vector<
+        std::future< std::vector<mpz_class> > 
```

Comment:
I'm going to change this before I accept this. this amount of splitting really hurts readability, IMO.